bootstrap/app.php

```diff
@@ -6,12 +6,12 @@
 
 return Application::configure(basePath: dirname(__DIR__))
     ->withRouting(
-        web: __DIR__.'/../routes/web.php',
-        commands: __DIR__.'/../routes/console.php',
+        web: __DIR__ . '/../routes/web.php',
+        commands: __DIR__ . '/../routes/console.php',
         health: '/up',
     )
     ->withMiddleware(function (Middleware $middleware) {
-        //
+        $middleware->redirectGuestsTo(fn() => abort(404));
```

Comment:
Nice! This is a very elegant solve.

Reply:
I wonder whether there's a way to customize the 404 page, but I don't think it's necessary right now other than to satisfy our curiosity.